mapApp/static/mapApp/js/index.js

```diff
@@ -63,43 +85,53 @@ if (typeof zoom !== 'undefined') {
     locateUser(setView = true, watch = false);
 }
 
-// Add geocoder control
-var geocoder = L.Control.geocoder({
-    position: "topleft",
-    placeholder: gettext('Search...'),
-    errorMessage: gettext('Nothing found.')
-}).addTo(map);
-var geocodeMarker;
-geocoder.markGeocode = function (result) {
-    map.fitBounds(result.bbox);
-    geocodeMarker && map.removeLayer(geocodeMarker); //remove old marker if it exists
-
-    geocodeMarker = new L.Marker(result.center, {
-        icon: icons["geocodeIcon"]
-    }).bindPopup(result.name).addTo(map).openPopup();
-};
-
-// Add scalebar
-L.control.scale({
-    position: 'bottomright'
-}).addTo(map);
-
-/* Turn off default mousewheel event (map zoom in / zoom out) when mouse is over the legend. This allows scrolling when contents overflow legend div */
-var elem = L.DomUtil.get('legend');
-L.DomEvent.on(elem, 'mousewheel', L.DomEvent.stopPropagation);
-
-// Add all points to map
-//collisions = geojsonMarker(collisions, "collision").addTo(incidentData).getLayers(),
-//nearmisses = geojsonMarker(nearmisses, "nearmiss").addTo(incidentData).getLayers(),
-//hazards = geojsonMarker(hazards, "hazard").addTo(incidentData).getLayers(),
-//thefts = geojsonMarker(thefts, "theft").addTo(incidentData).getLayers();
-//newInfrastructures = geojsonMarker(newInfrastructures, "newInfrastructure").addTo(incidentData).getLayers();
+/**
+* Data fetching
+*/
+
+/**
+ * @type {Object[]} - Array of objects
+ * @type {string} Object[].id - The type of incident (eg collision, hazard)
+ * @type {layer} Object[].layer - leaflet layer generated from geojson data for one incident type
+ */
+ var incidentLayers = [];
+
+/**
+* Group of all the incident layers together.
+* Marker cluster group similar to or extension of layergroup?
+*/
+var incidentData = new L.MarkerClusterGroup({
+    maxClusterRadius: 70,
+    polygonOptions: {
+        color: '#2c3e50',
+        weight: 3
+    },
+    animateAddingMarkers: true,
+    iconCreateFunction: pieChart
+});
 
+incidentData.addTo(map);
 
-//add in the xhr layers here
+// Create data feature groups
+var collisions, nearmisses, hazards, thefts, newInfrastructures;
+loadIncidentDataByType("/nearmisses_tiny?format=json", "nearmiss", nearmisses);
```

Comment:
No need to change now, but if you end up refactoring the loadIncidentDataByType function, you could append ?format.json to the URL in that function instead of repeatedly adding it here.

Reply:
for sure, this is fixed in the refactored function in #72!